File: src/gestorAplicacion/serviciosExtra/Factura.java
package gestorAplicacion.serviciosExtra;

import java.io.Serializable;
import java.util.Arrays; //static import


public class Factura implements Serializable{

    private int FacturaHospedaje;
    private int FacturaServiciosExtra;
    private int FacturaRestaurante;
    private int FacturaVehiculo;
    private int[] idsServicios = new int[3];

    public String mostrarListaServiciosExtra(){
        return """
            ========Servicios disponible tiene valor extra en la factura==========\n
            1. Gimnasio\n
            2. Bar\n
            3. Parqueadero\n
                """;
    }

    public boolean seleccionarServicio(int id){
        ServiciosExtra servicio = ServiciosExtra.buscarPorId(id);
        FacturaServiciosExtra += servicio.getPrecio();
        return true;
    }

    public int CalcularPrecioFactura(){
        return FacturaHospedaje + FacturaServiciosExtra + FacturaRestaurante + FacturaVehiculo;
    }

    public boolean contains(final int[] arr, final int key) {
        return Arrays.stream(arr).anyMatch(i -> i == key);
    }

    @Override
    public String toString() {
        return String.format("""
            Factura por hospedaje: %d
            Factura por servicios extra: %d
            Factura por restaurante: %d
            Factura por alquiler de vehiculo: %d
            """,FacturaHospedaje, FacturaServiciosExtra, FacturaRestaurante, FacturaVehiculo);
    }

    //* Getters y setteres */
    public int getFacturaHospedaje() {
        return FacturaHospedaje;
    }

    public void setFacturaHospedaje(int facturaHospedaje) {
        FacturaHospedaje = facturaHospedaje;
    }

    public int getFacturaServiciosExtra() {
        return FacturaServiciosExtra;
    }

    public void setFacturaServiciosExtra(int facturaServiciosExtra) {
        FacturaServiciosExtra = facturaServiciosExtra;
    }

    public int[] getIdsServicios() {
        return idsServicios;
    }

    public void setIdsServicios(int[] idsServicios) {
        this.idsServicios = idsServicios;
    }

    public int getFacturaRestaurante() {
        return FacturaRestaurante;
    }

    public void setFacturaRestaurante(int facturaRestaurante) {
        FacturaRestaurante = facturaRestaurante;
    }

    public int getFacturaVehiculo() {
        return FacturaVehiculo;
    }

    public void setFacturaVehiculo(int facturaVehiculo) {
        FacturaVehiculo = facturaVehiculo;
    }
    

    
}

File: python/src/gestor_aplicacion/servicios_extra/Factura.py
from gestor_aplicacion.servicios_extra.Servicios_extra import ServiciosExtra

class Factura:
    
    def __init__(self):
        self.FacturaHospedaje = 0
        self.FacturaServiciosExtra = 0
        self.FacturaRestaurante = 0
        self.FacturaVehiculo = 0
        self.idsServicios = [0] * 5

    def mostrarListaServiciosExtra(self):
        return '''
            ========Servicios disponible tiene valor extra en la factura==========
            1. Gimnasio
            2. Bar
            3. Parqueadero
        '''

    def seleccionarServicio(self, id):
        servicio = ServiciosExtra.buscarPorId(id)
        self.FacturaServiciosExtra += servicio.getPrecio()
        return True

    def CalcularPrecioFactura(self):
        return self.FacturaHospedaje + self.FacturaServiciosExtra + self.FacturaRestaurante + self.FacturaVehiculo

    def contains(self, arr, key):
        return key in arr

    def __str__(self):
        return '''
            Factura por hospedaje: {}
            Factura por servicios extra: {}
            Factura por restaurante: {}
            Factura por alquiler de vehiculo: {}
        '''.format(self.FacturaHospedaje, self.FacturaServiciosExtra, self.FacturaRestaurante, self.FacturaVehiculo)

    # Getters y setters
    def getFacturaHospedaje(self):
        return self.FacturaHospedaje

    def setFacturaHospedaje(self, facturaHospedaje):
        self.FacturaHospedaje = facturaHospedaje

    def getFacturaServiciosExtra(self):
        return self.FacturaServiciosExtra

    def setFacturaServiciosExtra(self, facturaServiciosExtra):
        self.FacturaServiciosExtra = facturaServiciosExtra

    def getIdsServicios(self):
        return self.idsServicios

    def setIdsServicios(self, idsServicios):
        self.idsServicios = idsServicios

    def getFacturaRestaurante(self):
        return self.FacturaRestaurante

    def setFacturaRestaurante(self, facturaRestaurante):
        self.FacturaRestaurante = facturaRestaurante

    def getFacturaVehiculo(self):
        return self.FacturaVehiculo

    def setFacturaVehiculo(self, facturaVehiculo):
        self.FacturaVehiculo = facturaVehiculo
